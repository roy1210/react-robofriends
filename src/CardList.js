import React from 'react';
import Card from './Card';

const CardList = ({ robots }) => {
  const cardComponents = robots.map((user, i) => {
    return (
      <Card
        id={robots[i].id}
        name={robots[i].name}
        email={robots[i].email}
        key={i}
      />
    );
  });
  return <div>{cardComponents}</div>;
};

export default CardList;
