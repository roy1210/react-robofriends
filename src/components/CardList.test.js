import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';
import '../setupTests';

it('renders CardList without crashing', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'John snow',
      username: 'John',
      email: 'john@gmail.com'
    }
  ];
  expect(shallow(<CardList robots={mockRobots} />).debug()).toMatchSnapshot();
});
