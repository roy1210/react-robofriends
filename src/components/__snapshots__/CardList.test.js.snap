// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CardList without crashing 1`] = `
"<div>
  <Card id={1} name=\\"John snow\\" email=\\"john@gmail.com\\" />
</div>"
`;
